example/generator/SinewaveLimitProvider.js

```diff
@@ -0,0 +1,88 @@
+/*****************************************************************************
+ * Open MCT, Copyright (c) 2014-2017, United States Government
+ * as represented by the Administrator of the National Aeronautics and Space
+ * Administration. All rights reserved.
+ *
+ * Open MCT is licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * Open MCT includes source code licensed under additional open source
+ * licenses. See the Open Source Licenses file (LICENSES.md) included with
+ * this source code distribution or the Licensing information page available
+ * at runtime from the About dialog for additional information.
+ *****************************************************************************/
+/*global define*/
+
+define([
+
+], function (
+
+) {
+
+    var RED = 0.9,
+        YELLOW = 0.5,
+        LIMITS = {
+            rh: {
+                cssClass: "s-limit-upr s-limit-red",
```

Comment:
Hrm, should our limit evaluators be defining CSS classes? I kind of expect limit evaluators to simply determine whether a given telemetry value is out of limits, and for the view to determine how to visually represent this.

Reply:
Concepts such as "red high" and "yellow low" are not universal, as other projects may have "ERROR", "OK" with different display needs.  Additionally, LimitEvaluators can be repurposed as "value stylers" for color coding or other purposes.  Allowing developers to define custom cssClasses allows them to achieve whatever styling they desire.  Plots are an edge case where limitEvaluators influence alarm markers and ignore the specific cssClass; if/when that becomes a problem we will be better equipped to work it out.